presto-orc/src/main/java/com/facebook/presto/orc/reader/StructSelectiveStreamReader.java

```diff
@@ -751,4 +766,78 @@ public long getRetainedSizeInBytes()
             return INSTANCE_SIZE + sizeOf(outputPositions);
         }
     }
+
+    private static final class MissingFieldStreamReader
+            implements SelectiveStreamReader
+    {
+        private static final int INSTANCE_SIZE = ClassLayout.parseClass(MissingFieldStreamReader.class).instanceSize();
+        private final Type outputType;
+        private int[] outputPositions;
+        private int outputPositionCount;
+
+        MissingFieldStreamReader(Optional<Type> type)
+        {
+            this.outputType = type.isPresent() ? type.get() : null;
```

Comment:
Is there any reason for making an instance of MissingFieldStreamReader if output type is missing? I think not. If so, let's change the constructor to take Type (not Optional<Type>) and add requireNotNull.